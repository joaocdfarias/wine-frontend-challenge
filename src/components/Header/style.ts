import Link from 'next/link'
import styled from 'styled-components'

interface ActiveLinkProps {
  isCurrentPage: boolean
}

export const HeaderStyled = styled.header`
  width: 100%;
  background: #fff;
  padding: 16px;
  box-shadow: 0px 2px 5px 0px #cccccc;

  .content {
    display: flex;
    align-items: center;
    justify-content: space-between;
  }

  .left {
    display: flex;
    align-items: center;
    gap: 84px;
  }

  img {
    max-width: 103px;
  }

  ul {
    display: flex;
  }

  .links ul {
    gap: 48px;
  }

  .options ul {
    gap: 42px;
  }

  .options li {
    position: relative;
  }
`
export const ActiveLink = styled.a<ActiveLinkProps>`
  color: ${(props) => (props.isCurrentPage ? '#D14B8F' : '#555555')};
  position: relative;
  font-size: 18px;
  font-weight: 400;

  &::after {
    content: '';
    width: 100%;
    height: 2px;
    position: absolute;
    bottom: -34px;
    background-color: #d14b8f;
    display: ${(props) => (props.isCurrentPage ? 'block' : 'none')};
  }

  cursor: pointer;
`

export const CounterStyled = styled.span`
  width: 20px;
  height: 20px;
  background-color: #ffffff;
  color: #4fbfa5;
  position: absolute;
  right: -5px;
  bottom: 5px;
  box-shadow: 0px 2px 5px 0px #cccccc;
  text-align: center;
  border-radius: 40px;
  font-size: 16px;
  font-weight: 700;
`
