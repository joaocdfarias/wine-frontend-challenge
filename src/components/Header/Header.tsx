import React from 'react'
import Image from 'next/image'
import * as S from './style'
import Link from 'next/link'

const headerLinks = [
  {
    name: 'Clube',
    link: '/clube',
  },
  {
    name: 'Loja',
    link: '/',
  },
  {
    name: 'Produtores',
    link: '/produores',
  },
  {
    name: 'Ofertas',
    link: '/ofertas',
  },
  {
    name: 'Eventos',
    link: '/eventos',
  },
]

const optionsIcons = [
  { name: 'Busca', url: '/search.svg' },
  { name: 'Conta', url: '/account.svg' },
  { name: 'Wine Box', url: '/wine-box.svg', hasCounter: true },
]

export const Header = () => {
  const [isCurrentPage, setIsCurrentPage] = React.useState<string>('')
  const [cartCounter, setCartCounter] = React.useState<number>(0)

  React.useEffect(() => {
    const currentPage = window.location.pathname

    setIsCurrentPage(currentPage)
  }, [])

  return (
    <S.HeaderStyled>
      <div className="container content">
        <div className="left">
          <Image src="/logo.svg" alt="Wine Logo" width="103px" height="31px" />
          <div className="links">
            <ul>
              {headerLinks.map((item) => {
                return (
                  <li key={item.name}>
                    <Link href={item.link}>
                      <S.ActiveLink isCurrentPage={isCurrentPage === item.link}>{item.name}</S.ActiveLink>
                    </Link>
                  </li>
                )
              })}
            </ul>
          </div>
        </div>
        <div className="right">
          <div className="options">
            <ul>
              {optionsIcons.map((icon) => {
                return (
                  <li key={icon.name}>
                    <Link href="/">
                      <Image src={icon.url} alt={icon.name} width="56px" height="56px" />
                    </Link>
                    {icon.hasCounter && <S.CounterStyled>0</S.CounterStyled>}
                  </li>
                )
              })}
            </ul>
          </div>
        </div>
      </div>
    </S.HeaderStyled>
  )
}
