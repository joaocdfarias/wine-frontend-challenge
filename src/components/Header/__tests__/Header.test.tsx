import { render } from '@testing-library/react'
import { Header } from '../Header'

describe('Header', () => {
  it('should render correctly', () => {
    const { baseElement } = render(<Header />)

    expect(baseElement).toMatchSnapshot()
  })
})
