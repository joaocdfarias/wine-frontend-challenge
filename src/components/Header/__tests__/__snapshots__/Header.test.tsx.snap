// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<body>
  <div>
    <header
      class="sc-cb7c5640-0 uErJu"
    >
      <div
        class="container content"
      >
        <div
          class="left"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27103%27%20height=%2731%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="Wine Logo"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
          <div
            class="links"
          >
            <ul>
              <li>
                <a
                  class="sc-cb7c5640-1 cwtgqG"
                >
                  Clube
                </a>
              </li>
              <li>
                <a
                  class="sc-cb7c5640-1 enXcJI"
                >
                  Loja
                </a>
              </li>
              <li>
                <a
                  class="sc-cb7c5640-1 cwtgqG"
                >
                  Produtores
                </a>
              </li>
              <li>
                <a
                  class="sc-cb7c5640-1 cwtgqG"
                >
                  Ofertas
                </a>
              </li>
              <li>
                <a
                  class="sc-cb7c5640-1 cwtgqG"
                >
                  Eventos
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="right"
        >
          <div
            class="options"
          >
            <ul>
              <li>
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2756%27%20height=%2756%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Busca"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </li>
              <li>
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2756%27%20height=%2756%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Conta"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </li>
              <li>
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2756%27%20height=%2756%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Wine Box"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
                <span
                  class="sc-cb7c5640-2 hraBxq"
                >
                  0
                </span>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </header>
  </div>
</body>
`;
